### PR TITLE
fix(reprocessing): Add indexes on datetime cols for reprocessing models to fix `clear_expired_raw_events`

### PR DESCRIPTION
**Pull request by wedamija**
The `clear-expired-raw-events` task has been failing for a while: https://sentry.sentry.io/crons/sentry/clear-expired-raw-events/?environment=prod&statsPeriod=24h

This is because it performs batch deletes on these tables using `datetime`, but there's no index available

Also limit the maximum number of times the query can run, and increase the batch size to 1000.

### CODE REVIEW

File: src/sentry/tasks/reprocessing.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def clear_expired_raw_events():
     from sentry.models.rawevent import RawEvent
     from sentry.models.reprocessingreport import ReprocessingReport
 
+    MAX_BATCHES_PER_MODEL = 1000
```
Comment (evanpurkhiser) [inline src/sentry/tasks/reprocessing.py:64]: Does this mean the script won't remove everything in one go then?

Comment (wedamija) [reply src/sentry/tasks/reprocessing.py:64]: Yeah, but it runs every 15 minutes. It's 1000 batches of 1000 deletions, so 1 milliion every 15 minutes. The `sentry_rawevent` table has around 50m events, so even that will only take a short while to clean up.

Also I'm going to wait until next week to merge this so it's not running over the weekend

Comment (evanpurkhiser) [reply src/sentry/tasks/reprocessing.py:64]: 👍 

Will let processing team look at this as well. Looks good though

File: src/sentry/tasks/reprocessing.py [hunk 2]
```diff
@@ -72,10 +74,10 @@ def batched_delete(model_cls, **filter):
         # out at that point and never do the delete.
         #
         # Better to delete a few rows than none.
-        while True:
+        for _ in range(MAX_BATCHES_PER_MODEL):
             # Django already loads this into memory, might as well do it
             # explicitly. Makes check for result emptiness cheaper.
-            result = set(model_cls.objects.filter(**filter)[:200].values_list("pk", flat=True))
+            result = set(model_cls.objects.filter(**filter)[:1000].values_list("pk", flat=True))
```
Comment (anonrig) [inline src/sentry/tasks/reprocessing.py:80]: Is `[:1000]` similar to SQL's limit? ORM makes it extremely hard to understand the database implications of this change

Comment (anonrig) [reply src/sentry/tasks/reprocessing.py:80]: Also shouldn't `:1000` be replaced with `MAX_BATCHES_PER_MODEL`?

Comment (wedamija) [reply src/sentry/tasks/reprocessing.py:80]: Yeah it just adds a `LIMIT 1000`. Fetching/deleting 1000 ids at a time should be fine here

Comment (wedamija) [reply src/sentry/tasks/reprocessing.py:80]: > Also shouldn't `:1000` be replaced with `MAX_BATCHES_PER_MODEL`?

No, that's the max number of time to make the query. The numbers happen to be the same but are separate concepts

Comment (anonrig) [reply src/sentry/tasks/reprocessing.py:80]: I see. Thank you for explaining. Would you mind adding a comment?

**Review by anonrig — APPROVED** [review]

**Pull request merged**